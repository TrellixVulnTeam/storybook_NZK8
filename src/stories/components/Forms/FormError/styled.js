// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { inputErrorFontStyle } from '../FormStyles';

export const Wrapper = styled.label`
  font-size: 14px;
  display: block;
  padding: 5px 0 0 2px;
  ${inputErrorFontStyle}
`;
