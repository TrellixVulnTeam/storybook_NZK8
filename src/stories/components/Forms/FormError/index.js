// @flow

import React from 'react';
import { Wrapper } from './styled';

type Props = {
  children: React$Node,
};

/**
 * Basic form field error that shows below most
 * input fields when a validation error is present
 * @example <FormError>Email is invalid</FormError>
 * @param {Object} props
 * @return {React$Node}
 */
export default (props: Props) => <Wrapper>{props.children}</Wrapper>;
