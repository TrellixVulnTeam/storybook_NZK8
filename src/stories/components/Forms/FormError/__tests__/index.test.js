import React from 'react';
import { shallow, mount } from 'enzyme';
import FormError from '../index';
import 'jest-styled-components';

describe('<FormError />', () => {
  it('snapshot', () => {
    const c = shallow(<FormError />);

    expect(c).toMatchSnapshot();
  });

  it('should render props.children', () => {
    const c = mount(<FormError>Invalid Email</FormError>);

    expect(c.find('label').text()).toContain('Invalid Email');
  });
});
