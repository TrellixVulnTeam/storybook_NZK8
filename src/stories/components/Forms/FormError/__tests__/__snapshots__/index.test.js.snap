// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormError /> snapshot 1`] = `<styled__Wrapper />`;
